waspc/data/Generator/templates/server/package.json

```diff
@@ -4,9 +4,10 @@
   "version": "0.0.0",
   "private": true,
   "type": "module",
+  "comment-filip": "The server.js location changed because we have now included client source files above .wasp/out/server/src.",
   "scripts": {
     "build": "npx tsc",
-    "start": "npm run validate-env && NODE_PATH=dist node -r dotenv/config dist/server.js",
+    "start": "npm run validate-env && NODE_PATH=dist node -r dotenv/config dist/.wasp/out/server/src/server.js",
```

Comment:
Them writing the code anywhere -> no, I don't think its a problem!

Ok, so TS in `dist/` now when generating code starts from the top of Wasp project so it also pulls in this extra structure of .wasp/out/... . Makes sense I guess! A bit weird that package.json has to know this structure but ok I guess? We could in theory inject this knowledge into package.json via variables from Haskell maybe? Or maybe it doesn't matter and this is ok hardcoded like this.

Reply:
It doesn't need to know anything about TS's file structure (at least not directly).

All it needs to know is what script to execute to run the server (which is influenced by TS's file structure).

I think it can stay hardcoded. It's harcoded in the current system too, but for a different path.

Reply:
One last thought about this: consider local NPM packages (the same thing we use for the SDK): https://github.com/wasp-lang/wasp/issues/1429#issuecomment-1702578721

Reply:
I think we discussed live that it might make sense to not have this hardcoded at the end, right? Or do I remember it incorrectly?

Reply:
Related: https://github.com/wasp-lang/wasp/issues/1769#issuecomment-1952938100